### PR TITLE
[18Rhl] Complete implementation of all company abilities

### PR DESCRIPTION
**Pull request by perwestling**
Some minor changes was needed in general code to cover the case when
a player owned company performs a tokening - as the tokener is
then not the comany, or company owner, but the current entity
(corporation) that is operating. This should not affect existing
titles, as a check is done if owner is player.

### CODE REVIEW

File: lib/engine/game/g_18_mex/game.rb [hunk 1]
```diff
@@ -69,7 +69,7 @@ class Game < Game::Base
           '470' =>
           {
             'count' => 1,
-            'color' => 'yellow',
+            'color' => 'orange',
```
Comment (tobymao) [inline lib/engine/game/g_18_mex/game.rb:72]: 18mex?

File: lib/engine/game/g_18_rhl/step/lay_tile_checks.rb [hunk 2]
```diff
@@ -5,5 +5,13 @@ def lay_tile(action, extra_cost: 0, entity: nil, spender: nil)
     super
 
     @game.eastern_ruhr_connection_check(action.tile.hex)
+    return if action.tile.hex.name != 'E14' || !@game.prinz_wilhelm_bahn
+
+    # Remove hex block
+    comp = @game.prinz_wilhelm_bahn
+    comp.all_abilities.each do |a|
```
Comment (tobymao) [inline lib/engine/game/g_18_rhl/step/lay_tile_checks.rb:12]: do you need to dup here or no

File: lib/engine/step/special_token.rb [hunk 3]
```diff
@@ -9,6 +9,7 @@ class SpecialToken < Base
       include Tokener
 
       def actions(entity)
+        return [] unless ability(entity)
```
Comment (tobymao) [inline lib/engine/step/special_token.rb:12]: why did you add this, it's the same as line 13

File: lib/engine/step/tokener.rb [hunk 4]
```diff
@@ -33,6 +33,8 @@ def token_cost_override(_entity, _city_hex, _slot, _token)
 
       def available_tokens(entity)
         token_holder = entity.company? ? entity.owner : entity
+        return [] if token_holder.player?
```
Comment (tobymao) [inline lib/engine/step/tokener.rb:36]: is this correct?

File: lib/engine/game/g_18_mex/game.rb (hunk 1 above)
Comment (perwestling) [reply lib/engine/game/g_18_mex/game.rb:72]: Yeah, 18MEX has a similar orange special tile, but it broke 18MEX so that can be added in a separate commit. Maybe.
Will add a commit that removes this change.

File: lib/engine/step/special_token.rb (hunk 3 above)
Comment (perwestling) [reply lib/engine/step/special_token.rb:12]: Right - removed.

File: lib/engine/step/tokener.rb (hunk 4 above)
Comment (perwestling) [reply lib/engine/step/tokener.rb:36]: Does not seem to be needed after all, so I have now removed this change.

File: lib/engine/game/g_18_rhl/step/lay_tile_checks.rb (hunk 2 above)
Comment (perwestling) [reply lib/engine/game/g_18_rhl/step/lay_tile_checks.rb:12]: Probably safe in this case as it only contains one ability... but sure adding dup is fine, so I am doing that.
